astropy/coordinates/angles.py

```diff
@@ -644,12 +650,20 @@ def _wrap_internal(self):
         # Convert the wrap angle and 360 degrees to the native unit of
         # this Angle, then do all the math on raw Numpy arrays rather
         # than Quantity objects for speed.
+        # TODO: This could probably be sped up as well.  u.degree.to(self.unit)
```

Comment:
@embray - I may not have been clear: I'd prefer to remove the if statement on l. 659 for `self.wrap_angle.unit`, but think the one on l.666, which checks whether wrapping is necessary before attempting it should stay since it seems to have a much larger benefit.